frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -219,19 +236,17 @@ function FileManagerMenu:setUpdateItemTable()
             return G_reader_settings:readSetting("lastfile") ~= nil
         end,
         callback = function()
+            self:openLastDoc()
+        end,
+        hold_callback = function()
             local last_file = G_reader_settings:readSetting("lastfile")
-            if not last_file or lfs.attributes(last_file, "mode") ~= "file" then
-                local InfoMessage = require("ui/widget/infomessage")
-                UIManager:show(InfoMessage:new{
-                    text = _("Cannot open last document"),
-                })
-                return
-            end
-            local ReaderUI = require("apps/reader/readerui")
-            ReaderUI:showReader(last_file)
-            self:onCloseFileManagerMenu()
-            local FileManager = require("apps/filemanager/filemanager")
-            FileManager.instance:onClose()
+            UIManager:show(ConfirmBox:new{
+                text = T(_("Would you like to open the last document: %1?"), last_file),
+                ok_text = _("OK"),
```

Comment:
Isn't that the default? I'd say this should be `Open` but it's not entirely clear to me what the purpose of this ConfirmBox is. Could you please explain the rationale?

Reply:
It's the default for `callback`: open last doc without any feedback.
This is for `hold_callback`, the confirmbox is just some intermediate step to show the filename, and decide to go on or not.

Reply:
Exactly :)
+ callback (normal tap) - open last document without any feedback (as it was)
+ hold_callback (long tap) - open confirm box (new)

Reply:
It offers the user an answer to the "what do I get, using this menu item", part of the "inline user manual" the new approach to the Long Press protocol can provide.  Great! One down, many more to go.